### PR TITLE
[RFC] tests: migrate legacy test 11

### PR DESCRIPTION
**Pull request by lucc**
There is something strange going on in this test.  At some points I had to add "empty vim commands" in order to stop the test from stalling.  And at one point even that did not help.  Does anyone have an idea about that?

### CODE REVIEW

**Comment by watiko** [comment]
@lucc Maybe the same issue (except for `FilterRead{Pre,Post}`) happened to us. (#3877)
To avoid the issue I used `helpers.feed()`.

```lua
-- helpers.execute('command`)
helpers.feed(':command<CR><C-l>')
```

And `FilterRead{Pre,Post}` has another issue. I replaced some `execute()` with one `source()` then I got the error `Vim(!):E493: Backwards range given: 23,$!cat`.

I hope this is helpful for us :)

**Comment by lucc** [comment]
@watiko thanks for these comments. I think I fixed it.

**Comment by lucc** [comment]
ref: part of #1286

**Comment by justinmk** [comment]
@lucc Shouldn't `testdir/test11.*` be deleted (and `testdir/Makefile` updated)?

File: test/functional/legacy/011_spec.lua [hunk 1]
```diff
@@ -0,0 +1,231 @@
...
+-- being modified outside of Vim (noticed on Solaris).
+
+local helpers, lfs = require('test.functional.helpers'), require('lfs')
+local clear, execute, expect, eq, neq, dedent, write_file =
+  helpers.clear, helpers.execute, helpers.expect, helpers.eq, helpers.neq,
+  helpers.dedent, helpers.write_file
+local wait = helpers.wait -- only for debugging
+
+local function has_gzip()
+  return os.execute('gzip --help >/dev/null 2>&1') == 0
```
Comment (justinmk) [inline test/functional/legacy/011_spec.lua]: `helpers.eval('executable("gzip")') == 1`

File: test/functional/legacy/011_spec.lua [hunk 2]
```diff
@@ -0,0 +1,231 @@
...
+      os.execute('gzip --force Xtestfile')
+      -- This should create Xtestfile.gz and delete Xtestfile.
+      neq(nil, lfs.attributes('Xtestfile.gz'))
+      eq(nil, lfs.attributes('Xtestfile'))
+    end
+  end)
+  teardown(function()
+    os.remove('Xtestfile.gz')
+    os.remove('Xtest.c')
+--    os.remove('foo.c')
```
Comment (justinmk) [inline test/functional/legacy/011_spec.lua]: is this needed?

File: test/functional/legacy/011_spec.lua [hunk 3]
```diff
@@ -0,0 +1,231 @@
...
+      --execute('au FileChangedShell * echo "caught FileChangedShell"')
+      execute('set bin')
+      execute("au FileReadPost    *.gz   '[,']!gzip -d")
+      -- Read and decompress the testfile.
+      execute('$r Xtestfile.gz')
+      wait()
+      expect('\n'..text1)
+    end)
+
+    it('BufRead{Pre,Post} (using gzip)', function()
```
Comment (justinmk) [inline test/functional/legacy/011_spec.lua]: if it fits on one line, writing it out like `BufReadPre,BufReadPost` makes grepping easier.

**Comment by justinmk** [comment]
LGTM after comments and squash.

**Comment by lucc** [comment]
I also moved the zipping of Xtestfile into the respective test block (and out of setup()) and moved one more test block into the `if gzip then test else pending end` block.

**Comment by justinmk** [comment]
Validated. Thanks!

**Pull request merged**